common/djangoapps/third_party_auth/views.py

```diff
@@ -110,3 +112,14 @@ def post_to_custom_auth_form(request):
         'hmac': pipeline_data['hmac'],
     }
     return render(request, 'third_party_auth/post_custom_auth_entry.html', data)
+
+
+class IdPRedirectView(View):
+    def get(self, request, *args, **kwargs):
+        next_destination_url = get_next_url_for_login_page(request)
+
+        try:
+            url = pipeline.get_login_url(kwargs['provider_slug'], pipeline.AUTH_ENTRY_LOGIN, next_destination_url)
+            return redirect(url)
+        except ValueError:
```

Comment:
@davestgermain Looking at this again, would it be a good idea to log something here?

Reply:
Maybe, but this is fine for now.